### ISSUE TITLE
BUG: agent stuck in an infinite loop, while trying to discover K8S custom resources

### ISSUE THREAD

**Issue by xdvpser** [scalyr/scalyr-agent-2#998]
We installed `scalyr-agent` on our k8s cluster through Helm, but it wasn't working as expected. While testing, we found a bug in Scalyr Agent that caused high CPU usage and logs not being sent. We suspect that it gets stuck in an endless loop while trying to perform k8s objects discovery. Specifically, it was stuck on our Redis pods. Agent was trying to discover its controllers, but was going into infinite loop when encountering unknown Redis k8s custom resource. Following lines were repeatedly appearing in debug logs (`/var/log/scalyr-agent-2/agent_debug.log` file):
```
2022-09-21 10:46:08.569Z Level 7 [core] [scalyr_agent.monitor_utils.k8s:775] cache value for Redis default/redis: <scalyr_agent.monitor_utils.k8s.Controller object at 0x7fa2e1e1f7f0> (allow_expired=True)
2022-09-21 10:46:08.569Z Level 7 [core] [scalyr_agent.monitor_utils.k8s:770] cache hit for Redis default/redis (allow_expired=True)
```
Notably, it works fine on k8s nodes where Redis pods are not running.
Default Agent configuration was used as provided by Helm chart.

Versions:
scalyr-agent - `2.1.33`
helm chart - `0.2.19`

**Comment by Kami** [comment]
Resolved via #999 and #1004.

**Comment by Kami** [comment]
v2.1.37 stable release with this bug fix is now available - https://github.com/scalyr/scalyr-agent-2/releases/tag/v2.1.37.

Thanks again for your contribution.
